src/base/bittorrent/session.cpp

```diff
@@ -3531,10 +3539,21 @@ void Session::handleTorrentShareLimitChanged(TorrentHandle *const torrent)
     updateSeedingLimitTimer();
 }
 
-void Session::saveTorrentResumeData(TorrentHandle *const torrent, bool finalSave)
+void Session::handleTorrentsPrioritiesChanged()
+{
+    // Save fastresume for the torrents that changed queue position
+    for (TorrentHandle *const torrent : torrents()) {
+        if (!torrent->isSeed()) {
+            if (torrent->queuePosition() != torrent->nativeHandle().queue_position()) // cached vs actual queue position
```

Comment:
Thankfully i tested it and we forgot we returned queuepos +1 in `queuePosition()` 
so i'll have to `torrent->nativeHandle().queue_position() + 1` here and
`resumeData["qBt-queuePosition"] = nativeHandle().queue_position() + 1;`
The question is wouldn't that be more confusing for a reader? I'll do them if you think it's better.

Reply:
Or make a new function `actualQueuePosition()` or something?

Reply:
>The question is wouldn't that be more confusing for a reader?

It's OK for me, personally, since it's used in abstraction layer internals only.
>Or make a new function actualQueuePosition() or something?

It's more confusing to have this method in TorrentHandle interface. This raises questions, like "Why are we using no actual data to somewhere else?"

Maybe just add appropriate commits in affected places?

Reply:
>Maybe just add appropriate commits in affected places?

I don't know what's an appropriate commit for this, or you meant comment?

Reply:
>or you meant comment?

Oops... Yes, I meant comment. Sorry.

Reply:
Done.